aws/handler_functions.go

```diff
@@ -59,6 +59,9 @@ func ValidateResponseHandler(r *Request) {
 	}
 	r.Retryable = r.Service.ShouldRetry(r)
 	r.RetryDelay = r.Service.RetryRules(r)
+	if r.Retryable {
+		r.ResetReaderBody()
```

Comment:
I'm thinking this logic might be bettered suited to a BeforeRetry handler.  We already have AfterRetry and Retry handlers so I'm wondering if in this case it would make sense to create a BeforeRetry. What do you think?